Makefile

```diff
@@ -70,22 +70,31 @@ help: ## Display this help.
 .PHONY: generate
 generate: ## Generate boilerplate DeepCopy methods, manifests, and Go client
 	# Use uid and gid of current user to avoid mismatched permissions
-	iidfile=$$(mktemp /tmp/iid-XXXXXX) && \
+	set -e ; \
+	iidfile=$$(mktemp /tmp/iid-XXXXXX) ; \
 	docker build \
 		--build-arg USER_ID=$(shell id -u $(USER)) \
 		--build-arg GROUP_ID=$(shell id -g $(USER)) \
 		--build-arg CONTROLLER_TOOLS_VERSION=$(CONTROLLER_TOOLS_VERSION) \
 		--build-arg CODE_GENERATOR_VERSION=$(CODE_GENERATOR_VERSION) \
 		--file neonvm/hack/Dockerfile.generate \
-		--iidfile $$iidfile . && \
+		--iidfile $$iidfile . ; \
+	volumes=('--volume' "$$PWD:/go/src/github.com/neondatabase/autoscaling") ; \
+	if [ -f .git ]; then \
+		gitdir="$$(git rev-parse --git-common-dir)" ; \
+		volumes+=('--volume') ; \
+		volumes+=("$$gitdir:$$gitdir") ; \
+	fi ; \
```

Comment:
I probably lack some git knowledge, so I don't really understand this part. 
- In which case is `.git` a file and not a directory? Something related to worktrees? 
- We mount `$PWD` as `/go/src/github.com/neondatabase/autoscaling` (and set workdir it also), won't it mount also `.git`?
- `git rev-parse --git-common-dir` may return a relative path (for a regular git repo, and an absolute one for a submodule), but a relative path can't be used for volume (I guess I might understand it once I get answers for  previous questions)

Reply:
Yeah...

> In which case is `.git` a file and not a directory? Something related to worktrees?

Yeah, if you're in a git worktree, `.git` is a file with contents like
```
gitdir: /path/to/common/dir/worktrees/name
```

> We mount `$PWD` as `/go/src/github.com/neondatabase/autoscaling` (and set workdir it also), won't it mount also `.git`?

So yeah, if you're using worktrees, then the git _directory_ (stored elsewhere) won't be included.

For example, my `main` worktree is located at `/home/em/neon/autoscaling/main`, with the `.git` _file_ at `/home/em/neon/autoscaling/main/.git`, but the git common dir (i.e. where the objects are stored) at `/home/em/neon/autoscaling/.git` (note: one directory up).

> `git rev-parse --git-common-dir` may return a relative path

ah crap, really? :disappointed:

Reply:
Ah, got it, thanks for the explanation!

> > `git rev-parse --git-common-dir` may return a relative path
> 
> ah crap, really? 😞

I've tried it only for a regular git repo (where `.git` is a directory): 
```
bayandin@mbp autoscaling % git rev-parse --git-common-dir
.git
```
Not sure about other cases

Reply:
Ok, added some handling for relative paths in 4122115dad2f9434af1e2037f17d0ffb0ab99101